logstash-core/src/main/java/org/logstash/instrument/metrics/Metric.java

```diff
@@ -88,4 +59,16 @@ default String type() {
         return getType().asString();
     }
 
+    /**
+     * Determine if this metric has a value explicitly set.
+     * @return true if this metric has been set to a specific value, false if it is the default state
+     */
+    boolean isDirty();
```

Comment:
Actually, thinking through this, we really do want to change this behavior.

Writing a client for the current interface is problematic because some stats don't exist, then suddenly do. A pipeline, for instance, should always have its events counter active at `0` even if it hasn't been explicitly set.

I'm wondering if there's a way to compare the behavior here for testing purposes while changing this functionality?